src/settingsView.ts

```diff
@@ -316,12 +319,34 @@ function htmlForWebview(vscode: vscodeTypes.VSCode, extensionUri: vscodeTypes.Ur
           div.appendChild(label);
           projectsElement.appendChild(div);
         }
+
+        const allEnabled = projects.every(p => p.enabled);
+        if (projects.every(p => p.enabled)) {
+          selectAllCheckbox.checked = true;
+          selectAllCheckbox.indeterminate = false;
+        }
+        else if (projects.some(p => p.enabled)) {
+          selectAllCheckbox.indeterminate = true;
+        }
+        else {
+          selectAllCheckbox.checked = false;
+          selectAllCheckbox.indeterminate = false;
+        }
       }
 
+      selectAllCheckbox.addEventListener('change', event => {
+        for (const checkbox of projectsElement.querySelectorAll('input[type=checkbox]')) {
+          checkbox.checked = selectAllCheckbox.checked;
+          checkbox.dispatchEvent(new Event('change'));
+        }
+      });
+
       const vscode = acquireVsCodeApi();
       for (const input of document.querySelectorAll('input[type=checkbox]')) {
         input.addEventListener('change', event => {
-          vscode.postMessage({ method: 'toggle', params: { setting: event.target.getAttribute('setting') } });
+          const setting = event.target.getAttribute('setting');
+          if (setting)
```

Comment:
What happens if there's no setting (I don't know this VSCode API)?

Reply:
It's standard DOM. We have some input elements in the HTML that have a setting attribute, e.g. this:

https://github.com/Skn0tt/playwright-vscode/blob/9f109ef3ce9af487b00136196fc67810cb0e1ee2/src/settingsView.ts#L276

Before this change, there were no elements without a setting. But i've added a new `input` element that doesn't have it, so we need to check this here. Makes me think we shouldn't attach the event listener in the first place, letb me change that!

Reply:
adding it to the css selector is even smoother I think: https://github.com/microsoft/playwright-vscode/pull/569/commits/a1e1a8258eba61aaaa48e9fbafccd84b6eedc9e6